src/acl/mock.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  WithResourceInfo,
+  WithResourceAcl,
+  WithAcl,
+  WithAccessibleAcl,
+  WithFallbackAcl,
+  UrlString,
+} from "../interfaces";
+import { getSourceIri } from "../resource/resource";
+import { createAcl, internal_getContainerPath } from "./acl";
+import { mockContainerFrom } from "../resource/mock";
+
+/**
+ * Function for use in unit tests to mock a [[SolidDataset]] that has its own ACL attached.
+ *
+ * Warning: do not use this function in actual production code.
+ * This function initialises a new empty ACL and attaches it to a given [[SolidDataset].
+ * This is useful to mock a SolidDataset with an ACL in tests of code that call e.g.
+ * [[getResourceAcl]].
+ *
+ * @param resource The Resource that we should pretend has its own ACL.
+ * @returns The input Resource, with an empty Resource ACL attached.
+ * @since Not released yet.
+ */
+export function addMockResourceAclTo<T extends WithResourceInfo>(
+  resource: T
+): T & WithResourceAcl {
+  const aclUrl =
+    resource.internal_resourceInfo.aclUrl ?? getSourceIri(resource) + ".acl";
```

Comment:
```suggestion
  // The ACL URL being the resource URL with a .acl prefix is just for readability, 
  // but it is not a convention that should be relied on. ACL discovery is described in
  // the Solid specification
  const aclUrl =
    resource.internal_resourceInfo.aclUrl ?? getSourceIri(resource) + ".acl";
```

I'm afraid that not having a warning might lead some people looking at the code to jump to conclusions. I'm unsure whether the warning should include a link to the new spec, as it is currently a draft, to the old spec, wich will soon be outdated, or remain as it is there.

Reply:
Yeah good point. I think it might be best to just use an obviously arbitrary URL, rather than basing it off the source URL.

Reply:
bcbfd70767b29c2ad74ace146b4d83048ddfab6d